### DISCUSSION TITLE
Help with GRE in IPSEC

### DISCUSSION THREAD

**Discussion by Neeqza** [strongswan/strongswan#2485] [Q&A] [answered]
Hi,

I'm trying to configure GRE in IPsec using strongswan and I'm having issues getting it to work.  The IPsec tunnel is up, but I can't ping the tunnels endpoints. IPsec peers are servers with public IP addresses (privates addresses on the config files are stand-ins)

Here is a snippet of swanctl.conf : 

`connections {
    ubuntu {
        local_addrs=172.16.10.10
        remote_addrs=192.168.10.10
        encap=yes
        mobike=no

        local {
                auth=psk
                id=172.16.10.10
        }
        remote {
                auth=psk
                id=192.168.10.10
        }
        children {
                gre {
                   esp_proposals=aes256gcm16
                   local_ts=dynamic[gre]
                   remote_ts=dynamic[gre]
                   mode=transport
                   start_action=start
                 }
        }
    version=0
    proposals=default
  }

}

secrets {
  ike-ubuntu {
    id=192.168.10.10
    secret=azerty
    }
}
`

GRE Interface : 

`gre1@NONE: <POINTOPOINT,NOARP,UP,LOWER_UP> mtu 1476 qdisc noqueue state UNKNOWN group default qlen 1000
    link/gre 172.16.10.10 peer 192.168.10.10
    inet 10.64.40.95/24 scope global gre1
`

**Comment by tobiasbrunner** [comment]
Routes? Firewall rules? What are the actual endpoints for the ping?

**Reply by Neeqza** [reply]
> Routes? Firewall rules? What are the actual endpoints for the ping?

On server A

Routes : 
`192.168.131.0/24 dev gre1 scope link`

FW: 
`iptables -A FORWARD -o gre1 -j ACCEPT
iptables -A FORWARD -i gre1 -j ACCEPT
`
Hosts would be 192.168.128.10/24 behind server A and 192.168.131.10/24 Behind server B.

Edit : 

GRE endpoints are 
`10.62.40.96  ; 10.62.40.95`

IP forwarding is on.

Essentially the same on the other side.

**Reply by tobiasbrunner** [reply]
> GRE endpoints are
> `10.62.40.96 ; 10.62.40.95`

I guess you replaced those in your original post? Or why are the addresses different there? If not, make sure the interface and strongSwan config match.

And if you want to find out where packets get stuck, try to look at the various stats (on the interface, the firewall rules, the IPsec SAs etc.).

**Comment by Neeqza** [comment, marked as answer]
Resolved. It was necessary to link the GRE interface to a physical interface using `ip link add [NAME] type gre remote [IP] local [IP] dev [INTERFACE]`

**Reply by tobiasbrunner** [reply]
Hm, interesting. This isn't necessary in our [testing environment](https://strongswan.org/testing/testresults/route-based/net2net-gre/). Maybe because we add the GRE interface via `ip tunnel add` and not `ip link add`?

**Reply by Neeqza** [reply]
I also added the tunnel using `ip tunnel add`. I re did it using netplan (without the link) and it works fine for now. It seems weirdly fickle when I don't directly link interfaces so it might be some routing problems on my end.
